cmd/asset-syncer/postgresql_utils.go

```diff
@@ -62,24 +69,40 @@ func (m *postgresAssetManager) Sync(charts []models.Chart) error {
 }
 
 func (m *postgresAssetManager) initTables() error {
-	_, err := m.DB.Exec(fmt.Sprintf(
-		"CREATE TABLE IF NOT EXISTS %s (ID serial NOT NULL PRIMARY KEY, chart_id varchar unique, info jsonb NOT NULL)",
-		dbutils.ChartTable,
-	))
+	// Repository table should have a namespace column, and chart table should reference repositories.
+	_, err := m.DB.Exec(fmt.Sprintf(`
+CREATE TABLE IF NOT EXISTS %s (
+	ID serial NOT NULL PRIMARY KEY,
+	namespace varchar NOT NULL,
+	name varchar NOT NULL,
+	checksum varchar,
+	last_update varchar,
+	UNIQUE(namespace, name)
+)`, dbutils.RepositoryTable))
 	if err != nil {
 		return err
 	}
-	_, err = m.DB.Exec(fmt.Sprintf(
-		"CREATE TABLE IF NOT EXISTS %s (ID serial NOT NULL PRIMARY KEY, name varchar unique, checksum varchar, last_update varchar)",
-		dbutils.RepositoryTable,
-	))
+
+	_, err = m.DB.Exec(fmt.Sprintf(`
+CREATE TABLE IF NOT EXISTS %s (
+	ID serial NOT NULL PRIMARY KEY,
+	repository_id integer NOT NULL REFERENCES %s (ID),
+	chart_id varchar,
+	info jsonb NOT NULL,
+	UNIQUE(repository_id, chart_id)
+)`, dbutils.ChartTable, dbutils.RepositoryTable))
 	if err != nil {
 		return err
 	}
-	_, err = m.DB.Exec(fmt.Sprintf(
-		"CREATE TABLE IF NOT EXISTS %s (ID serial NOT NULL PRIMARY KEY, chart_files_ID varchar unique, info jsonb NOT NULL)",
-		dbutils.ChartFilesTable,
-	))
+
+	_, err = m.DB.Exec(fmt.Sprintf(`
+CREATE TABLE IF NOT EXISTS %s (
+	ID serial NOT NULL PRIMARY KEY,
+	namespace varchar NOT NULL,
+	chart_files_ID varchar NOT NULL,
+	info jsonb NOT NULL,
+	UNIQUE(namespace, chart_files_ID)
+)`, dbutils.ChartFilesTable))
```

Comment:
check the function `RepoAlreadyProcessed` below, I think that require to use the namespace as well

Reply:
Yes, as does Delete. I plan to update those in a later PR (it doesn't affect things when reposPerNamespace is false).